### PR TITLE
[Tests and Docs] Add a test on serializing pipelines with components containing fused LoRA modules

### PR DESCRIPTION
**Pull request by sayakpaul**
This PR adds:

* a test ensuring pipelines with components containing fused LoRA modules can be serialized correctly. 
* documentation 

This has been quite a requested feature from the community I think. 

@DN6 okay for you if we added this test?

### CODE REVIEW

**Review by patrickvonplaten — APPROVED** [review]
Cool!

**Comment by DN6** [comment]
@sayakpaul  If you don't mind just adding the total time taken to run these LoRA tests here as a comment? Just to have an idea of the additional overhead being introduced to the runners? The pytest summary report is fine.

**Review by stevhliu — APPROVED** [review]
Nice work! 👍

**Comment by sayakpaul** [comment]
> @sayakpaul If you don't mind just adding the total time taken to run these LoRA tests here as a comment? Just to have an idea of the additional overhead being introduced to the runners? The pytest summary report is fine.

@DN6 find it below. 

[logs_58445.zip](https://github.com/huggingface/diffusers/files/12578490/logs_58445.zip)

**Comment by sayakpaul** [comment]
I thought https://github.com/huggingface/diffusers/pull/4971 fixed the FLAX problem. @patrickvonplaten?

File: docs/source/en/training/lora.md [hunk 1]
```diff
@@ -333,7 +333,48 @@ pipe.fuse_lora(lora_scale=0.5)
...
+pipe = DiffusionPipeline.from_pretrained("stabilityai/stable-diffusion-xl-base-1.0", torch_dtype=torch.float16).to("cuda")
+lora_model_id = "hf-internal-testing/sdxl-1.0-lora"
+lora_filename = "sd_xl_offset_example-lora_1.0.safetensors"
+pipe.load_lora_weights(lora_model_id, weight_name=lora_filename)
+
+# First, fuse the LoRA parameters.
+pipe.fuse_lora()
+
+# Then save.
+pipe.save_pretrained("my-pipeline-with-fused-lora", safe_serialization=True)
```
Comment (patrickvonplaten) [inline docs/source/en/training/lora.md:364]:
```suggestion
pipe.save_pretrained("my-pipeline-with-fused-lora")
```

this is done automatically

**Comment by patrickvonplaten** [comment]
Only failing test seems to be the flaky one. Let's merge this

**Pull request merged**